### PR TITLE
Feature adapter/transform class renaming

### PR DESCRIPTION
**Pull request by gokulavasan**
Transform Interface -> Transformation Interface
TransformStage -> Transform
StageContext -> TransformContext
Removed Property

Build : http://builds.cask.co/browse/CDAP-RT20-1

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java [hunk 1]
```diff
@@ -17,29 +17,19 @@
...
-  public boolean isRequired() {
-    return required;
-  }
+public interface Transformation<IN, OUT> {
 
-  public String getDescription() {
-    return description;
-  }
+  /**
+   * Process input and emit output using {@link Emitter}.
```
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java]: Fix docs
`Transform the input and emit output using {@link Emitter}

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java [hunk 2]
```diff
@@ -17,29 +17,19 @@
...
+   *
+   * @param input the input to transform
```
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java]: Fix docs
`@param input data to be transformed

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java [hunk 3]
```diff
@@ -17,29 +17,19 @@
...
+   * @param emitter {@link Emitter} to emit data to the next stage
+   * @throws Exception
```
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java]: Document when does it throw Exception

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java [hunk 4]
```diff
@@ -17,29 +17,19 @@
 package co.cask.cdap.templates.etl.api;
 
 /**
- * Properties that are required by a Stage.
+ * Transforms an input object into zero or more output objects.
```
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java]: How can it be more than one? If you say split a string into array - it is still one object which is of type array right?

File: cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/templates/etl/common/TrackedTransform.java [hunk 5]
```diff
@@ -18,20 +18,20 @@
 
 import co.cask.cdap.api.metrics.Metrics;
 import co.cask.cdap.templates.etl.api.Emitter;
-import co.cask.cdap.templates.etl.api.Transform;
+import co.cask.cdap.templates.etl.api.Transformation;
 
 /**
- * A {@link Transform} that delegates transform operations to another Transform while emitting metrics
+ * A {@link Transformation} that delegates transform operations to another Transform while emitting metrics
```
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/templates/etl/common/TrackedTransform.java]: Fix docs - remove "to another Transform" i feel it is redundant.

**Comment by sreevatsanraman** [comment]
Minor java doc fixes. Please fix and merge

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java (hunk 4 above)
Comment (gokulavasan) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transformation.java]: Oh, using Emitter you can emit multiple objects. So there could be 1->Many.

**Comment by gokulavasan** [comment]
Build passed : http://builds.cask.co/browse/CDAP-RT20-4. Merging.

**Pull request merged**